sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/transform/BeamBuiltinAggregations.java

```diff
@@ -347,4 +357,35 @@ public BigDecimal toBigDecimal(BigDecimal record) {
       return record;
     }
   }
+
+  static class BitOr<T extends Number> extends CombineFn<T, BitOr.Accum, Long> {
+    static class Accum {
```

Comment:
I am thinking there is no need to define a class to wrap `long val`? Just use `Long`?

Also you could consider this interface: https://github.com/apache/beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Sum.java#L129?

Reply:
Hmm, Im not sure if this works. How would I create the accumulator? I can't do `new Long()`. That's why I wrapped `long` in Accum

Reply:
Actually, this might work, let me test this
```
  static class BitOr<T extends Number> extends CombineFn<T, Long, Long> {
    @Override
    public Long createAccumulator() {
      return 0L;
    }

    @Override
    public Long addInput(Long accum, T input) {
      return accum | input.longValue();
    }

    @Override
    public Long mergeAccumulators(Iterable<Long> accums) {
      Long merged = createAccumulator();
      for (Long accum : accums) {
        merged = merged | accum;
      }
      return merged;
    }

    @Override
    public Long extractOutput(Long accum) {
      return accum;
    }
  }
}
```

Reply:
It worked! Will commit this now!